--- a/lib/Slic3r.pm	2017-12-01 12:56:32.000000000 -0500
+++ b/lib/Slic3r.pm	2017-12-04 15:36:56.174577313 -0500
@@ -41,7 +41,7 @@
 use FindBin;
 
 # Let the XS module know where the GUI resources reside.
-set_resources_dir(decode_path($FindBin::Bin) . (($^O eq 'darwin') ? '/../Resources' : '/resources'));
+set_resources_dir(decode_path($FindBin::RealBin) . "/../share/slic3r-prusa3d" . (($^O eq 'darwin') ? '/../Resources' : '/resources'));
 set_var_dir(resources_dir() . "/icons");
 set_local_dir(resources_dir() . "/localization/");
  
