--- a/CMakeLists.txt	2017-11-22 21:23:00.656956898 -0500
+++ b/CMakeLists.txt	2017-11-22 21:26:34.038774760 -0500
@@ -41,6 +41,7 @@
 message("PATH: $ENV{PATH}")
 message("PERL5LIB: $ENV{PERL5LIB}")
 find_package(Perl REQUIRED)
+find_package(PerlLibs REQUIRED)
 
 add_subdirectory(xs)
 
@@ -53,5 +54,5 @@
 
 file(GLOB MyVar var/*.png)
 install(FILES ${MyVar} DESTINATION share/slic3r-prusa3d)
-install(FILES lib/Slic3r.pm DESTINATION lib/slic3r-prusa3d)
-install(DIRECTORY lib/Slic3r DESTINATION lib/slic3r-prusa3d)
+install(FILES lib/Slic3r.pm DESTINATION ${PERL_VENDORLIB})
+install(DIRECTORY lib/Slic3r DESTINATION ${PERL_VENDORLIB})
--- a/xs/CMakeLists.txt	2017-11-22 21:23:00.656956898 -0500
+++ b/xs/CMakeLists.txt	2017-11-22 21:26:34.038774760 -0500
@@ -549,5 +549,5 @@
 endif ()
 
 # Installation
-install(TARGETS XS DESTINATION lib/slic3r-prusa3d/auto/Slic3r/XS)
-install(FILES lib/Slic3r/XS.pm DESTINATION lib/slic3r-prusa3d/Slic3r)
+install(TARGETS XS DESTINATION ${PERL_VENDORARCH}/auto/Slic3r/XS)
+install(FILES lib/Slic3r/XS.pm DESTINATION ${PERL_VENDORLIB}/Slic3r)
